### PR TITLE
Add new nat gateway resource

### PR DESCRIPTION
**Pull request by khannarheams**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by maddieclayton** [comment]
@azuresdkci Add to whitelist

**Comment by maddieclayton** [comment]
@khannarheams Can you link the design review for this change? https://github.com/Azure/azure-powershell-cmdlet-review-pr

**Comment by khannarheams** [comment]
@maddieclayton  https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/279

File: src/Network/Network/Models/PSSubnet.cs [hunk 1]
```diff
@@ -27,6 +27,9 @@ public class PSSubnet : PSChildResource
         [JsonProperty(Order = 1)]
         public List<PSIPConfiguration> IpConfigurations { get; set; }
 
+        [JsonProperty(Order = 1)]
+        public PSNatGateway NatGateway { get; set; }
```
Comment (EvgenyAgafonchikov) [inline src/Network/Network/Models/PSSubnet.cs]: Should this be
```
public PSResourceId NatGatewayId { get; set; }
```
to match specs?

Comment (khannarheams) [reply src/Network/Network/Models/PSSubnet.cs]: no, this should be similar to route table. We want to allow both id and resource

**Comment by khannarheams** [comment]
Updated to https://github.com/Azure/azure-powershell/pull/9008

**Pull request closed without merging**